datashape/discovery.py

```diff
@@ -15,7 +15,7 @@
                         Record, string, Null, DataShape, real, date_, time_,
                         Unit, timedelta_, TimeDelta, object_, String)
 from .predicates import isdimension, isrecord
-from .py2help import _strtypes, _inttypes
+from .py2help import _strtypes, _inttypes, MappingProxyType
```

Comment:
Not in scope here, but thoughts on using something like `six` rather than rolling our own py2/py3 compatibility module?